plugins/net_plugin/include/eos/net_plugin/message_buffer.hpp

```diff
@@ -151,6 +154,7 @@ namespace eosio {
      */
     std::vector<boost::asio::mutable_buffer> get_buffer_sequence_for_boost_async_read() {
       std::vector<boost::asio::mutable_buffer> seq;
+      FC_ASSERT(write_ind.first < buffers.size());
```

Comment:
This is a good assert, I have it locally and am hitting it.  It would be more useful if it printed some data out like the value of `write_ind.first` and `buffers.size()`

Also, the code that calls this doesn't properly catch and recover from thrown errors.  See net_plugin.cpp:1230